daemon/containerd/image_builder.go

```diff
@@ -402,21 +402,9 @@ func (i *ImageService) CreateImage(ctx context.Context, config []byte, parent st
 		exposedPorts[string(k)] = v
 	}
 
-	var ociHistory []ocispec.History
```

Comment:
So, I'm even wondering; could we do the reverse here? And instead of ` dimage.NewFromJSON(config)`, Unmashal to a type that contains OCIImage and V1 image? Something like;

```go
	type ociWithV1 struct {
		dimage.V1Image
		ocispec.Image
	}

	imgToCreate := &ociWithV1{}

	if err := json.Unmarshal(config, imgToCreate); err != nil {
		return nil, err
	}
```

Reply:
Hmm, maybe?  Would `json.Unmarshal` fill up both structs correctly in that case? (I've never tried something like that before :sweat_smile:)

It seems like maybe not?  https://go.dev/play/p/ZZuqVkWsZvF (`go vet` is also very angry about the overlapping `json` tags)